2008-12-12		<zorry@ume.nu>

	* gcc/varasm.c (decl_tls_model): Check flag_pic instead of flag_shlib.

	flag_pie never get use to change the tls model.	

--- gcc/varasm.c.zorry
+++ gcc/varasm.c
@@ -4614,7 +4614,7 @@
     }
 
   is_local = targetm.binds_local_p (decl);
-  if (!flag_shlib)
+  if (!flag_pic)
     {
       if (is_local)
 	kind = TLS_MODEL_LOCAL_EXEC;
